app/js/mainModule.js

```diff
@@ -1,42 +1,94 @@
-{ 
-  
-  const MainController = ($scope) => {
-    const invIndex = new InvertedIndex();
-    $scope.uploads = () => {
-      invIndex.readFile();
-    };
-    $scope.message = '';
-    $scope.searchResult =
+/**
+ * Instance of class created
+ * @type Object
+ */
+const invIndex = new InvertedIndex();
+/**
+ * Controller handling basic features of app
+ * @param  {Object} $scope Binds the view with the controller
+ * @return {Object}        [description]
+ */
+const MainController = ($scope) => {
+  $scope.message = '';
+  $scope.fileNames = [];
+  $scope.searchResult = {};
+  $scope.fileObjects = {};
+  $scope.allIndicies = {};
+  $scope.fileUpload = (event) => {
+    const allUploads = event.target;
+    for (let count = 0; count < allUploads.files.length; count += 1) {
+      invIndex.readFile(allUploads.files[count]).then((content) => {
+        if (JSON.parse(content)) {
+          $scope.fileObjects[allUploads.files[count].name] = JSON.parse(content);
+          if ((!$scope.fileNames
+            .includes(allUploads.files[count].name))) {
+            $scope.$apply($scope.fileNames.push(allUploads.files[count].name));
```

Comment:
you should pass a function to $scope.$apply, and run this in the function $scope.fileNames.push(allUploads.files[count].name)